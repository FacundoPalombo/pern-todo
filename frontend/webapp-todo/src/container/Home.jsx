import React from "react";
import actions from "./actions";
import { connect } from "react-redux";
import { Home } from "modules";
const stpSchema = {};
const dtpSchema = { ...actions };

const mapStateToProps = () => stpSchema;
const mapDispatchToProps = () => dtpSchema;
export const HomeContainer = connect(mapStateToProps, mapDispatchToProps)(Home);
