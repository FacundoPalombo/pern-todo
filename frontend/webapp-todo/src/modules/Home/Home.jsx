import React from "react";
import styles from "./Home.module.scss";

export const Home = ({ children }) => {
  return <main className={styles.main}>{children}</main>;
};
