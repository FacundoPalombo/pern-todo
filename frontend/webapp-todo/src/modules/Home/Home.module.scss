.main {
  width: 100vw;
  height: 100vw;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
}
