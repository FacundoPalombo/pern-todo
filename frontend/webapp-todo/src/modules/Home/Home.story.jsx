import React from "react";
import { Home } from "./Home";

export const withOutChildren = () => <Home />;
export const withChildren = () => (
  <Home>
    <h1>Hello world!</h1>
  </Home>
);

export default {
  title: "Home",
  component: Home,
};
