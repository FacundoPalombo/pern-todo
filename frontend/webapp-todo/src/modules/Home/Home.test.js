import React from "react";
import { render } from "@testing-library/react";
import { Home } from "./Home";

describe("GIVEN Home component", () => {
  const { asFragment } = render(<Home />);
  const firstRender = asFragment();

  it("THEN should render", () => {
    expect(firstRender).toMatchSnapshot();
  });
  it("THEN should render with children", () => {
    const { asFragment } = render(
      <Home>
        <div id="testSelector"></div>
      </Home>
    );
    expect(asFragment().querySelector("#testSelector")).not.toBeNull();
  });
});
