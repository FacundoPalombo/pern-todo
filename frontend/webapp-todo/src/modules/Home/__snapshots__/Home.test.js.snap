// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Home component THEN should render 1`] = `
<DocumentFragment>
  <main
    class="main"
  />
</DocumentFragment>
`;
